doc/sfputil/dump_sfp_eeprom.md

```diff
@@ -0,0 +1,156 @@
+# Dump SFP EEPROM page data in show techsupport command #
+
+## Table of Content
+
+### Revision
+
+### Scope
+
+SONiC show techsupport command provides the ability to collect system dump for debug purpose. Module EEPROM data is important information for PHY issue debugging, but it is not part of show techsupport command. This design will enhance show techsupport command to contain module EEPROM data.
+
+### Definitions/Abbreviations
+
+N/A
+
+### Overview
+
+1. Existing CLI `sfputil show eeprom-hexdump` shall be extended to dump EEPROM pages in a single call.
+2. Show techsupport command shall be extended to call `sfputil show eeprom-hexdump` command to collect module EEPROM data.
+
+### Requirements
+
+1. Show techsupport command shall collect module EEPROM raw data
+2. All existing cable types shall be supported except RJ45
+3. Vendor specific implementation is not required but still possible.
+
+### Architecture Design
+
+The current architecture is not changed
+
+### High-Level Design
+
+Submodule sonic-utilities shall be changed. By default, vendor platform API implementation is not required.
+
+#### sonic-utilities
+
+##### generate_dump change
+
+1. New subcommand `sfputil show eeprom-hexdump` shall be used to dump module EEPROM data
+2. EEPROM data shall be saved to path `$TARDIR/dump/`
+
+Sample code:
+```
+save_cmd "sfputil show eeprom-hexdump" "interface.xcvrs.eeprom.raw" &
+```
+
+Note: dumping EEPROM data might cause firmware busy, the command shall not be run parallel with save_saidump to avoid hardware access conflict.
```

Comment:
@Junchao-Mellanox can you elaborate firmware busy? What is the hardware access conflict?

Reply:
Dumping module EEPROM might need accessing certain register for a long time. save_saidump might access the same register. So, there could be a conflict.
Anyway, I will remove this "Note" as it is not common for all vendors.